// @asymmetrik/ngx-leaflet-draw/esm2020/lib/leaflet-draw.module.mjs
import { NgModule } from '@angular/core';
import { LeafletModule } from '@asymmetrik/ngx-leaflet';
import { LeafletDrawDirective } from './core/leaflet-draw.directive';
import * as i0 from "@angular/core";
export class LeafletDrawModule {
}
LeafletDrawModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.3", ngImport: i0, type: LeafletDrawModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LeafletDrawModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.3", ngImport: i0, type: LeafletDrawModule, declarations: [LeafletDrawDirective], imports: [LeafletModule], exports: [LeafletDrawDirective] });
LeafletDrawModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.3", ngImport: i0, type: LeafletDrawModule, imports: [[
            LeafletModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.3", ngImport: i0, type: LeafletDrawModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        LeafletModule
                    ],
                    exports: [
                        LeafletDrawDirective
                    ],
                    declarations: [
                        LeafletDrawDirective
                    ]
                }]
        }] });
